docs/fundamentals/startup.rst

```diff
@@ -1,8 +1,188 @@
-.. include:: /stub-topic.txt
+Application Startup
+===================
 
-|stub-icon| Application Startup
-===============================
+By `Steve Smith`_
 
-.. include:: /stub-notice.txt
+ASP.NET 5 provides complete control of how individual requests are handled by your application. The ``Startup`` class is the entry point to the application, setting up configuration and wiring up services the application will use. Developers configure a request pipeline in the ``Startup`` class that is used to handle all requests made to the application.
+
+In this article:
+	- `The Startup class`_
+	- `Understanding request delegates`_
+	- `Implementing middleware in its own class`_
+	- `Setting up your request pipeline`_
+	- `Run, Map, and Use`_
+	
+`Browse or download samples on GitHub <https://github.com/aspnet/Docs/tree/master/docs/fundamentals/startup/sample>`_.
+
+The Startup class
+-----------------
+
+In ASP.NET 5, the ``Startup`` class provides the entry point for an application. It's possible to have :doc:`environment-specific startup files and methods <environments>`, but regardless, one ``Startup`` class will serve as the entry point for the application. This ``Startup`` class can optionally accept dependencies in its constructor, such as an instance of ``IHostingEnvironment`` which can be used to customize the behavior of the class based on the environment in which it is running. Typically, the way an application will be configured is defined within its Startup class's constructor (learn more about :doc:`configuration`). The Startup class must define a ``Configure()`` method, and may optionally also define a ``ConfigureServices()`` method, which will be called by the server hosting the application when it is started.
```

Comment:
@davidfowl @glennc I'd like to see a crisper definition of what the convention is for Startup classes and how they are discovered and instantiated. Do they have to be public? Can they be in any namespace? Can they be in any assembly/package/project? Can they be static?

When is the constructor called relative to stuff happening at the hosting layer? How long does the Startup class instance live?

What services are available to be injected in the startup constructor? Can you add custom services?